pytest_operator/plugin.py

```diff
@@ -566,10 +594,8 @@ def cloud_name(self) -> Optional[str]:
     @property
     def keep_model(self) -> bool:
         """Represents whether the current model should be kept after tests."""
-        if self._init_keep_model:
-            return True
         current_state = self.current_alias and self._models.get(self.current_alias)
-        return current_state.keep if current_state else False
+        return current_state.keep if current_state else self._init_keep_model
```

Comment:
So long as the model exists, `track_model` will ensure the value of `keep` is set according to the spec.

If the current_alias is unset or the model isn't found -- fall back to the global flag.